src/MSBuild/XMake.cs

```diff
@@ -1365,12 +1365,19 @@ private static IEnumerable<BuildManager.DeferredBuildMessage> GetMessagesToLogIn
                         "MSBVersion",
                         ProjectCollection.DisplayVersion),
                     MessageImportance.Low),
-                new BuildManager.DeferredBuildMessage(
-                    ResourceUtilities.FormatResourceStringIgnoreCodeAndKeyword(
+            };
+
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBuildDebugEngine")))
```

Comment:
Would it be better to check `DebugUtils.DebugPath is not null`, so the env var stuff only has to happen once?

Reply:
I thought about that. DebugUtils.DebugPath is not null if MSBUILDDEBUGPATH is set even if MSBuildDebugEngine is not enabled, which I didn't like. Then it isn't a VS thing but also a thing people (or at least I) really do use from the command line, and a high priority message about a debug path that I know where is felt undesirable. That said, I should be able to reference the trait and get the best of both worlds:
```suggestion
            if (Traits.Instance.DebugEngine)
```

That's only risky if someone changes the environment variable at an awkward time, which shouldn't be important.